lib/Common/Memory/Recycler.cpp

```diff
@@ -4806,16 +4864,25 @@ Recycler::StartBackgroundMark(bool foregroundResetMark, bool foregroundFindRoots
     bool doBackgroundFindRoots = true;
     if (foregroundResetMark || foregroundFindRoots)
     {
-        RECYCLER_PROFILE_EXEC_BEGIN(this, Js::ResetWriteWatchPhase);
-        bool hasWriteWatch = (recyclerPageAllocator.ResetWriteWatch() && recyclerLargeBlockPageAllocator.ResetWriteWatch());
-        RECYCLER_PROFILE_EXEC_END(this, Js::ResetWriteWatchPhase);
-
-        if (!hasWriteWatch)
+        // REVIEW: SWB, if there's only write barrier page change, we don't scan and mark?
+#if ENABLE_WRITE_WATCH
+        if (!CONFIG_FLAG(ForceSoftwareWriteBarrier))
         {
-            // Disable concurrent mark
-            this->enableConcurrentMark = false;
-            return false;
+            if (!recyclerPageAllocator.ResetWriteWatch() || !recyclerLargeBlockPageAllocator.ResetWriteWatch())
```

Comment:
I'm confused by this- you call ResetWriteWatch here- and then you call it again two lines later. Is that intentional?